framework/src/play/src/main/scala/play/api/mvc/Headers.scala

```diff
@@ -18,6 +19,10 @@ import scala.collection.immutable.{ TreeMap, TreeSet }
  */
 class Headers(protected var _headers: Seq[(String, String)]) {
 
+  /** INTERNAL API: Content-Length or -1 if not known. */
+  private[play] def contentLength: Long =
+    get(HeaderNames.CONTENT_LENGTH).map(_.toLong).getOrElse(-1)
```

Comment:
You could consider adding special-casing for this into the `get` methods of `Headers`. i.e. bring some of your work up from your Akka wrapper into the superclass.